x-pack/plugins/cases/server/services/cases/index.ts

```diff
@@ -0,0 +1,1007 @@
+      throw error;
+    }
+  }
+
+  public async findCases({
+    client,
+    options,
+  }: FindCasesArgs): Promise<SavedObjectsFindResponse<ESCaseAttributes>> {
+    try {
+      this.log.debug(`Attempting to find cases`);
+      return await client.find<ESCaseAttributes>({
+        sortField: defaultSortField,
+        ...options,
+        type: CASE_SAVED_OBJECT,
+      });
+    } catch (error) {
+      this.log.error(`Error on find cases: ${error}`);
+      throw error;
+    }
+  }
+
+  public async findSubCases({
+    client,
+    options,
+  }: FindCasesArgs): Promise<SavedObjectsFindResponse<SubCaseAttributes>> {
+    try {
+      this.log.debug(`Attempting to find sub cases`);
+      // if the page or perPage options are set then respect those instead of trying to
+      // grab all sub cases
+      if (options?.page !== undefined || options?.perPage !== undefined) {
+        return client.find<SubCaseAttributes>({
+          sortField: defaultSortField,
+          ...options,
+          type: SUB_CASE_SAVED_OBJECT,
+        });
+      }
+
+      const stats = await client.find<SubCaseAttributes>({
+        fields: [],
+        page: 1,
+        perPage: 1,
+        sortField: defaultSortField,
+        ...options,
+        type: SUB_CASE_SAVED_OBJECT,
+      });
+      return client.find<SubCaseAttributes>({
+        page: 1,
+        perPage: stats.total,
+        sortField: defaultSortField,
+        ...options,
+        type: SUB_CASE_SAVED_OBJECT,
+      });
+    } catch (error) {
+      this.log.error(`Error on find sub cases: ${error}`);
+      throw error;
+    }
+  }
+
+  /**
+   * Find sub cases using a collection's ID. This would try to retrieve the maximum amount of sub cases
+   * by default.
+   *
+   * @param id the saved object ID of the parent collection to find sub cases for.
+   */
+  public async findSubCasesByCaseId({
+    client,
+    ids,
+    options,
+  }: FindSubCasesByIDArgs): Promise<SavedObjectsFindResponse<SubCaseAttributes>> {
+    if (ids.length <= 0) {
+      return {
+        total: 0,
+        saved_objects: [],
+        page: options?.page ?? defaultPage,
+        per_page: options?.perPage ?? defaultPerPage,
+      };
+    }
+
+    try {
+      this.log.debug(`Attempting to GET sub cases for case collection id ${ids.join(', ')}`);
+      return this.findSubCases({
+        client,
+        options: {
+          ...options,
+          hasReference: ids.map((id) => ({
+            type: CASE_SAVED_OBJECT,
+            id,
+          })),
+        },
+      });
+    } catch (error) {
+      this.log.error(
+        `Error on GET all sub cases for case collection id ${ids.join(', ')}: ${error}`
+      );
+      throw error;
+    }
+  }
+
+  private asArray(id: string | string[] | undefined): string[] {
+    if (id === undefined) {
+      return [];
+    } else if (Array.isArray(id)) {
+      return id;
+    } else {
+      return [id];
+    }
+  }
+
+  private async getAllComments({
```

Comment:
Doesn't need to be in this PR but I wonder if we should move some of these comment methods to the `Attachment` service eventually. I guess that'll make our services dependent on each other, but that's probably ok, as long as we don't have circular dependencies. It probably makes sense that the cases service needs some other services as well.

Reply:
I agree! I had the same thought but because some of those functions had sub cases logic inside them I decided to not touch them. I will take a note for it.